backend/uclapi/common/decorators.py

```diff
@@ -0,0 +1,381 @@
+import datetime
+from functools import wraps
+import re
+import redis
+
+from dashboard.models import App, TemporaryToken
+from dashboard.tasks import keen_add_event_task as keen_add_event
+
+from oauth.models import OAuthToken
+from oauth.scoping import Scopes
+
+from .helpers import PrettyJsonResponse as JsonResponse
+
+from uclapi.settings import REDIS_UCLAPI_HOST
+
+
+# Gets a variable from GET or POST not caring which is which
+def get_var(request, var_name):
+    if var_name in request.GET:
+        return request.GET[var_name]
+    elif var_name in request.POST:
+        return request.POST[var_name]
+    else:
+        return None
+
+
+class UclApiIncorrectDecoratorUsageException(Exception):
+    pass
+
+
+class UclApiIncorrectTokenTypeException(Exception):
+    pass
+
+
+def how_many_seconds_until_midnight():
+    """Returns the number of seconds until midnight."""
+    tomorrow = datetime.datetime.now() + datetime.timedelta(days=1)
+    midnight = datetime.datetime(
+        year=tomorrow.year, month=tomorrow.month,
+        day=tomorrow.day, hour=0, minute=0, second=0
+    )
+    return (midnight - datetime.datetime.now()).seconds
+
+
+def log_api_call(request, token, token_type):
+    service = request.path.split("/")[1]
+    method = request.path.split("/")[2]
+
+    headers = request.META
+    version_headers = {}
+    regex = re.compile("^HTTP_UCLAPI_.*_VERSION$")
+    for header in headers:
+        if regex.match(header):
+            version_headers[header] = headers[header]
+
+    queryparams = dict(request.GET)
+
+    if token_type == "oauth":

```

Comment:
The `parameters` dict for `general` and `oauth` are the same, can't they be merged?